### PR TITLE
Support for microphone level.

### PR DESCRIPTION
**Pull request by microbit-robert**

### CODE REVIEW

**Comment by microbit-robert** [comment]
@microbit-matt-hillsdon Let's discuss the comments in `microbithal_js.c` for the microphone functions to make sure I've not missed anything important.

File: src/board/ui.ts [hunk 1]
```diff
@@ -397,6 +404,65 @@ export class AccelerometerUI {
...
+  public soundLevel: RangeSensor = new RangeSensor(
+    "soundLevel",
+    0,
+    255,
+    0,
+    undefined
+  );
+  private lowThreshold: number;
+  private highThreshold: number;
+  private thresholdState: "lowThreshold" | "highThreshold" | "none";
```
Comment (microbit-robert) [inline src/board/ui.ts]: Unsure whether using `thresholdState` is the right approach.

Comment (microbit-matt-hillsdon) [reply src/board/ui.ts]: Rejigged to have access to previous and current values.

**Comment by microbit-matt-hillsdon** [comment]
I accidentally broke this because I didn't realise that 0,1 and 1,2 are both used to represent quiet/low and loud/high.
If [this PR is merged](https://github.com/microbit-foundation/micropython-microbit-v2/pull/109) then we can just bump MicroPython, otherwise we need a different conversion for the 0,1 case (set_threshold).

**Comment by microbit-matt-hillsdon** [comment]
> I accidentally broke this because I didn't realise that 0,1 and 1,2 are both used to represent quiet/low and loud/high. If [this PR is merged](https://github.com/microbit-foundation/micropython-microbit-v2/pull/109) then we can just bump MicroPython, otherwise we need a different conversion for the 0,1 case (set_threshold).

Put in a workaround on the assumption we'll merge this. If not I'll redo to have separate conversions.

**Pull request merged**